emission/analysis/classification/inference/labels/inferrers.py

```diff
@@ -118,12 +119,35 @@ def placeholder_predictor_demo(trip):
         return []
     random_user_input = random.choice(unique_user_inputs) if random.randrange(0,10) > 0 else []
 
-    logging.debug(f"In placeholder_predictor_demo: ound {len(unique_user_inputs)} for user {user}, returning value {random_user_input}")
+    logging.debug(f"In placeholder_predictor_demo: found {len(unique_user_inputs)} for user {user}, returning value {random_user_input}")
     return [{"labels": random_user_input, "p": random.random()}]
 
 # Non-placeholder implementation. First bins the trips, and then clusters every bin
 # See emission.analysis.modelling.tour_model for more details
 # Assumes that pre-built models are stored in working directory
 # Models are built using evaluation_pipeline.py and build_save_model.py
+# This algorithm is now DEPRECATED in favor of predict_cluster_confidence_discounting (see https://github.com/e-mission/e-mission-docs/issues/663)
 def predict_two_stage_bin_cluster(trip):
     return lp.predict_labels(trip)
+
+# Reduce the confidence of the clustering prediction when the number of trips in the cluster is small
+# See https://github.com/e-mission/e-mission-docs/issues/663
+def n_to_confidence_coeff(n, max_confidence=None, first_confidence=None, confidence_multiplier=None):
+    if max_confidence is None: max_confidence = 0.99  # Confidence coefficient for n approaching infinity -- in the GitHub issue, this is 1-A
+    if first_confidence is None: first_confidence = 0.80  # Confidence coefficient for n = 1 -- in the issue, this is B
+    if confidence_multiplier is None: confidence_multiplier = 0.30  # How much of the remaining removable confidence to remove between n = k and n = k+1 -- in the issue, this is C
+    return max_confidence-(max_confidence-first_confidence)*(1-confidence_multiplier)**(n-1)  # This is the u = ... formula in the issue
+
```

Comment:
later, I would like to put these into a config file. But definitely not needed now.